boards/posix/nrf52_bsim/Kconfig.board

```diff
@@ -14,6 +14,11 @@ config BOARD_NRF52_BSIM
 	# of it), so that the corresponding driver becomes available (see
 	# dependencies of the ENTROPY_NRF5_RNG option).
 	select HAS_HW_NRF_RNG
+	# Do the same for the CLOCK and POWER peripherals, so that the nrfx
+	# drivers for them can be used.
+	select HAS_HW_NRF_CLOCK
+	select HAS_HW_NRF_POWER
+	select HAS_NRFX
```

Comment:
Yes, and it seems that `HAS_CMSIS_CORE` should not be selected by `HAS_NRFX`. It will be selected by `CPU_CORTEX_M` when the build targets the real SoC: https://github.com/zephyrproject-rtos/zephyr/blob/202ca01dcbe760b825f550df4d8411ad09f34df0/arch/arm/core/aarch32/Kconfig#L10
and that should be sufficient IMO.

I'll address it in a separate PR.